internal/controller/pdbwatcher_controller.go

```diff
@@ -161,7 +170,27 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	// What will return us to the original state? Assuming we need a watch on PDB for it to come off
+	// Watch for changes in PDB to revert to original state
+	if pdb.Status.DisruptionsAllowed > 0 && *deployment.Spec.Replicas != pdbWatcher.Status.MinReplicas {
+		// Check if the resource version has changed
+		err = r.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: pdbWatcher.Namespace}, deployment)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if deployment.ResourceVersion != initialResourceVersion {
```

Comment:
the above get is not enough its not likely to have changed since you did the get further above. 
Instead wht you want is to save the resouce version to your crd whenever the crd is reconciled or when we updte it below with deployment.updte.

Reply:
This is still wrong. the get here will likely be the same you need to compare the deployments resourceversion to pdbWatcher.Status.ResourceVersion to see if anyone has changed lets talk about this tommrow. As is I don't think this is ever likely to be true.